src/wayland/xdg_activation/dispatch.rs

```diff
@@ -87,7 +87,7 @@ where
 {
     fn request(
         state: &mut D,
-        _: &Client,
+        client: &Client,
```

Comment:
I'd wonder would the `ClientId` be preferable or pass a whole `Client`? You can get the id from a `Client` but also do lookup on the `Client` you get.

Reply:
The original client could disconnect before the token is used by another client. I didn't want to force the object to keep on living. In that sense it serves as an `Option<Client>`.